pkg/framework/pod.go

```diff
@@ -0,0 +1,144 @@
+//
+// Provides builders and helper methods for preparing Pods and nested Containers
+//
+package framework
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+//
+// PodBuilder
+//
+type PodBuilder struct {
+	pod *v1.Pod
+}
+
+// NewPodBuilder Creates an instance of a PodBuilder helper
+func NewPodBuilder(name string, namespace string) *PodBuilder {
+	pb := new(PodBuilder)
+	pb.pod = new(v1.Pod)
+	pb.pod.Name = name
+	pb.pod.Namespace = namespace
+	pb.pod.Spec = v1.PodSpec{}
+	pb.pod.Status = v1.PodStatus{}
+	return pb
+}
+
+// NewContainerBuilder creates an instance of a ContainerBuilder helper
+func NewContainerBuilder(name string, image string) *ContainerBuilder {
+	cb := new(ContainerBuilder)
+	cb.c = v1.Container{}
+	cb.c.Name = name
+	cb.c.Image = image
```

Comment:
```
cb.c = v1.Container{Name: name, 
                                Image: image}
```
just an idea

Reply:
It had to be done this way because of the type composition nature of v1.Container.
Name and Image are inherited from other types, and cannot be set directly.